### PR TITLE
Towards merging of master branch with powheg_gcc530

### PR DESCRIPTION
**Pull request by perrozzi**
partial merge of master with powheg_gcc530 to highlight conflicts

### CODE REVIEW

**Comment by perrozzi** [comment]
@covarell here is the git (failed) merge. The syntax to look at is

<<<<<<< HEAD
  what is currently in master
=======
  what is currently in powheg_gcc530 and can't be merged automatically
>>>>>>> powheg_gcc530

we need to go one by one (some could be anyway trivial)

File: bin/Powheg/lheMacros/mergeLheFiles.cpp [hunk 1]
```diff
@@ -44,6 +44,7 @@ int main(int argc, char** argv)
       eventIt++;
     }	
 
+<<<<<<< HEAD
```
Comment (perrozzi) [inline bin/Powheg/lheMacros/mergeLheFiles.cpp]: this conflict is related to the change from 
</weightgroup>
to
</initrwgt>
I guess is due to the reweighting scheme adopted (i.e. reweight the entire LHE file or event by event) but I am not fully sure what to do

File: bin/Powheg/lheMacros/splitLheFiles.cpp [hunk 2]
```diff
@@ -30,7 +30,11 @@ int main(int argc, char** argv)
   bool writeEvent = true;
   bool writePdf = false;
   int eventIt = 0;
+<<<<<<< HEAD
```
Comment (perrozzi) [inline bin/Powheg/lheMacros/splitLheFiles.cpp]: in this case the "off-by-one" has been updated recently but, once more, I am not fully sure what to do

File: bin/Powheg/lheMacros/splitLheFiles.cpp [hunk 3]
```diff
@@ -39,7 +43,11 @@ int main(int argc, char** argv)
     
     if( line != "</LesHouchesEvents>" ) { 
       
+<<<<<<< HEAD
```
Comment (perrozzi) [inline bin/Powheg/lheMacros/splitLheFiles.cpp]: I guess this is also related to the change in the reweighting scheme

File: bin/Powheg/lheMacros/splitLheFiles.cpp [hunk 4]
```diff
@@ -48,7 +56,11 @@ int main(int argc, char** argv)
 	writeEvent = false;   writePdf = true;
       }
       
+<<<<<<< HEAD
```
Comment (perrozzi) [inline bin/Powheg/lheMacros/splitLheFiles.cpp]: I guess this is also related to the change in the reweighting scheme

File: bin/Powheg/run_lhe_parallel.sh [hunk 5]
```diff
@@ -3,12 +3,21 @@
 fail_exit() { echo "$@"; exit 1; }
 
 #set -o verbose
+<<<<<<< HEAD
```
Comment (perrozzi) [inline bin/Powheg/run_lhe_parallel.sh]: this change has to do with the additional parameter to pass the number of threads.
This script (run_lhe_parallel.sh)  should be used with NNLOPS locally, probably is irrelevant (but better to be consistent and propagate the number of threads)

File: bin/Powheg/run_lhe_parallel.sh [hunk 6]
```diff
@@ -25,10 +34,22 @@ echo "%MSG-POWHEG number of jobs = $numj"
 nume=${3}
 echo "%MSG-POWHEG number of events per job = $nume"
 
+<<<<<<< HEAD
```
Comment (perrozzi) [inline bin/Powheg/run_lhe_parallel.sh]: same as above

File: bin/Powheg/run_pwg.py [hunk 7]
```diff
@@ -97,8 +97,8 @@ def prepareJobForEvents (tag, i, folderName, EOSfolder) :
     prepareJob(tag, i, folderName)
 
     f = open (filename, 'a')
-    f.write ('cp -p ' + rootfolder + '/' + folderName + '/powheg.input ./' + '\n')
-    f.write ('cp -p ' + rootfolder + '/' + folderName + '/JHUGen.input ./' + '\n')
```
Comment (perrozzi) [inline bin/Powheg/run_pwg.py]: this is trivial as is equivalent, probably better to delete than comment out

File: bin/Powheg/lheMacros/mergeLheFiles.cpp (hunk 1 above)
Comment (perrozzi) [inline bin/Powheg/lheMacros/mergeLheFiles.cpp]: according to the blame feature
https://github.com/cms-sw/genproductions/blame/powheg_gcc530/bin/Powheg/lheMacros/mergeLheFiles.cpp
the change comes from a single commit (NNLOPS in new reweighting scheme) so is ok to accept initrwgt

**Comment by perrozzi** [comment]
accepted powheg_gcc530 changes for mergeLheFiles.cpp and run_lhe_parallel.sh

**Comment by perrozzi** [comment]
merging into the newly created branch called "master_powheg_gcc5" to be used for testing before to actually merge to master

**Comment by covarell** [comment]
Thanks a lot Luca! 
Indeed I will have a look at the branch before the final merge (too hard to follow all commits).
@yuanchao
